packages/exoflex/src/components/Switch.tsx

```diff
@@ -37,14 +60,14 @@ export default function Switch(props: SwitchProps) {
   } = props;
   let { colors, style: themeStyle } = useTheme();
   let [xValue] = useState(new Animated.Value(value ? 1 : 0));
-  let width = +(StyleSheet.flatten(trackStyle)?.width || widthProps);
+
+  let width = getTrueWidth('track', widthProps, [
+    themeStyle?.switch?.trackStyle,
+    trackStyle,
+  ]);
 
   let thumbSize =
-    +(
-      StyleSheet.flatten(thumbStyle)?.width ||
-      StyleSheet.flatten(themeStyle?.switch?.thumbStyle)?.width ||
-      width / 2
-    ) -
+    getTrueWidth('thumb', width, [themeStyle?.switch?.thumbStyle, thumbStyle]) -
```

Comment:
Ah right, the `defaultWidth` is not an actual default value constant.

Rather than dividing the width inside the function, I think it might be easier to understand if we just do it here:

```ts
let trackWidth = getTrueWidth(widthProp, someStyle);
let thumbSize = getTrueWidth(trackWidth / 2, someThumbStyle);
```

Wdyt?

Reply:
Sounds good to me! Dividing the trackWidth outside the helper function makes it more readable.